Packages/com.unity.inputsystem/InputSystem/Devices/InputDevice.cs

```diff
@@ -708,6 +709,55 @@ internal bool disabledWhileInBackground
         // NOTE: This contains *leaf* controls only.
         internal uint[] m_StateOffsetToControlMap;
 
+        // Holds the nodes that represent the tree of memory ranges that each control occupies. This is used when
+        // determining what controls have changed given a state event or partial state update.
+        internal ControlBitRangeNode[] m_ControlTreeNodes;
+
+        // An indirection table for control bit range nodes to point at zero or more controls. Indices are used to
+        // point into the m_ChildrenForEachControl array.
+        internal ushort[] m_ControlTreeIndices;
+
+        // When a device gets built from a layout, we create a binary tree from its controls where each node in the tree
+        // represents the range of bits that cover the left or right section of the parent range. For example, starting
+        // with the entire device state block as the parent, where the state block is 100 bits long, the left node will
+        // cover from bits 0-50, and the right from bits 50-100. For the left node, we'll get two more child nodes where
```

Comment:
nitpick: 0-50 and 51-99, because in 100 bit log state block bit indices are from 0 to 99 inclusive